include/triton/core/tritonbackend.h

```diff
@@ -1569,6 +1569,60 @@ TRITONBACKEND_ISPEC TRITONSERVER_Error* TRITONBACKEND_ModelBatchInitialize(
 TRITONBACKEND_ISPEC TRITONSERVER_Error* TRITONBACKEND_ModelBatchFinalize(
     void* userp);
 
+/// Get all information about an output tensor by name.  The tensor data is
+/// returned as the base pointer to the data and the size, in bytes, of the
+/// data. The caller does not own any of the returned values and must not
+/// modify or delete them. The lifetime of all returned values extends until
+/// 'response' is deleted.
+///
+/// \param response The response object.
+/// \param name The name of the output.
+/// \param datatype Returns the type of the output.
+/// \param shape Returns the shape of the output.
+/// \param dim_count Returns the number of dimensions of the returned
+/// shape.
+/// \param base Returns the tensor data for the output.
+/// \param byte_size Returns the size, in bytes, of the data.
+/// \param memory_type Returns the memory type of the data.
+/// \param memory_type_id Returns the memory type id of the data.
+/// \param userp The user-specified value associated with the buffer
+/// in TRITONSERVER_ResponseAllocatorAllocFn_t.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONBACKEND_ISPEC TRITONSERVER_Error*
+TRITONBACKEND_InferenceResponseOutputByName(
+    TRITONBACKEND_Response* response, const char* name,
+    TRITONSERVER_DataType* datatype, const int64_t** shape, uint64_t* dim_count,
+    const void** base, size_t* byte_size, TRITONSERVER_MemoryType* memory_type,
+    int64_t* memory_type_id, void** userp);
```

Comment:
I think the userp should be probably hidden from the backend APIs. @GuanLuo what do you think?

Reply:
Spoke with @GuanLuo, he said that there is already a function to obtain the buffer attributes so we should not only remove `userp`, but the whole list of buffer attributes including `byte_size`, `memory_type`, and `memory_type_id`. Will refactor.

Reply:
Yes, the `BufferAttributes` is supposed to be the struct to group tensor metadata like those.